src/Client/Updater.php

```diff
@@ -730,16 +743,35 @@ private function fetchAndVerifyTargetsMetadata(string $role): void
         $targetsVersion = $newSnapshotData->getFileMetaInfo("$role.json")['version'];
         $newTargetsContent = $this->fetchFile("$targetsVersion.$role.json");
         $newTargetsData = TargetsMetadata::createFromJson($newTargetsContent, $role);
-        // TUF-SPEC-v1.0.12 Section 5.4.1
+        // TUF-SPEC-v1.0.13 Section 5.5.1
         $newSnapshotData->verifyNewHashes($newTargetsData);
-        // TUF-SPEC-v1.0.12 Section 5.4.2
+        // TUF-SPEC-v1.0.13 Section 5.5.2
         $this->checkSignatures($newTargetsData);
-        // TUF-SPEC-v1.0.12 Section 5.4.3
+        // TUF-SPEC-v1.0.13 Section 5.5.3
         $newSnapshotData->verifyNewVersion($newTargetsData);
-        // TUF-SPEC-v1.0.12 Section 5.4.4
-        static::checkFreezeAttack($newTargetsData, $this->getCurrentTime());
+        // TUF-SPEC-v1.0.13 Section 5.5.4
+        static::checkFreezeAttack($newTargetsData, $this->getMetadataExpirationTime());
         $newTargetsData->setIsTrusted(true);
-        // TUF-SPEC-v1.0.12 Section 5.4.5
+        // TUF-SPEC-v1.0.13 Section 5.5.5
         $this->durableStorage["$role.json"] = $newTargetsContent;
     }
+
+    /**
+     * Sets the time after which metadata files should be considered expired.
+     *
+     * This should only be called once during ::refresh() and should not be called
+     * otherwise.
+     *
+     * @throws \Tuf\Exception\FormatException
+     */
+    protected function setMetadataExpiration()
+    {
+        if ($this->isRefreshed) {
+            throw new \LogicException('::setMetadataExpiration() cannot be called after refresh has been called()');
+        } elseif ($this->metadataExpiration) {
+            throw new \LogicException('::setMetadataExpiration() should only be called once during ::refresh()');
+        }
```

Comment:
I put this logic into to avoid this ever being called 2x unless you refresh. I also don't see why you could want to call this outside of refresh.

It thought about  refactoring this into `metadataExpirationCanBeset(): bool` or `validateMetadataExpirationSet()` but that seemed a bit much.

right now to override and retain this logic you have to:

```
protected function setMetadataExpiration()
    {
        parent::setMetadataExpiration();
        $this->metadataExpiration = $this->getMetadataExpirationTime()->add(\DateInterval::createFromDateString('1 day'));
    }
```

Reply:
One thing I'd prefer is not to force the subclasses to set a specific property; I'd rather that they could return a DateTime object which the updater remembers. So something like:

```
protected function computeMetadataExpiration()
{
  return parent::computeMetadataExpiration()->add(\DateInterval::createFromDateString('1 day'));
}
```

Reply:
I pushed up a change with different name. let me know what you think